### PR TITLE
add simple URL runnable example

### PR DESCRIPTION
**Pull request by schimmy**
Optimus is sufficiently complex that we should have a few runnable examples to show off the power and provide a guide to new users.

I've added a `examples` directory and this is the first simple example. I'd also like to add (in separate PRs):
- group by
- join
- writing to s3 without using disk
- sort, map, & reduce (MapReduce on the cheap! although still in memory :/)

### CODE REVIEW

File: examples/simple_url.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+3) write the data to JSON
+4) output the number of rows we read in and the number of rows after filtering
+
+It's possible make an even simpler example, but this is a little more interesting.
+*/
+
+var (
+	inputFilename  = filepath.Join("examples", "urls.csv")
+	outputFilename = filepath.Join("examples", "urls.json")
+	cleverRegex, _ = regexp.Compile("clever.com.*")
```
Comment (natebrennand) [inline examples/simple_url.go]: I think you want to use [`regexp.MustCompile`](https://golang.org/pkg/regexp/#MustCompile) here

File: examples/simple_url.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+// helper function to keep things clean
+func fatalIfErr(err error) {
+	if err != nil {
+		log.Fatalf("error in URL example! Message is: %s", err)
+	}
+}
+
+// filterItems will only output SSL=true URL=/clever.com.*/ entries
+// if SSL or URL are missing, don't bother erroring and instead drop the row
+func filterItems(row optimus.Row) (bool, error) {
```
Comment (azylman) [inline examples/simple_url.go]: Can we name this after what it does? Something like `selectOnlyCleverSSL`

File: examples/simple_url.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	url, ok := row["url"]
+	if !ok || !cleverRegex.MatchString(url.(string)) {
```
Comment (azylman) [inline examples/simple_url.go]: Do you think it would be better to use `url.Parse` and check the host value instead of a regex?

**Comment by schimmy** [comment]
I would like to push this, and figure out the godoc later. I don't have time right now to figure out this stuff (even though I'm sure it's not too hard), and I do want the gophercon people to have an example they can run when they are reviewing the proposal!

The test failure is due to it being on a branch and the example package not existing yet, this all passes locally.

**Comment by azylman** [comment]
@schimmy It looks like your example doesn't run anymore because it doesn't have a main function. Also, it breaks the tests. After those are fixed you can merge.

**Comment by schimmy** [comment]
reverted to the earlier version - I'll push this and then make runnable in the future

**Pull request merged**